main.go

```diff
@@ -61,6 +73,10 @@ func main() {
 
 }
 
+// healthCheck is a http handler for health check to make sure the server is up.
 func healthCheck(w http.ResponseWriter, r *http.Request) {
-	w.Write([]byte("we are live"))
+	_, err := w.Write([]byte("we are live"))
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Are you sure you want to kill the server just because you couldn't write a response to a client?

Reply:
Why? this is super unsuffisticated.

Reply:
If a client connection gets dropped or any other client error you don't want to take down the whole instance due to a bad client. The instance could still be in a good state with other requests in flight.

Reply:
I would either ignore the error with an `_` or just log a message, I wouldn't exit. Given that this is a health check endpoint, it's up to whatever is running the app to decide what to do if it gets an error response back from the app.